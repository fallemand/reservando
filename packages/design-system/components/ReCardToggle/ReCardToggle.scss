@import "../../styles/variables";

.re-card-toggle {
  &__switch {
    flex-direction: row-reverse;
  }

  &__content {
    margin-top: $bdu * 2;
  }

  .re-switch__label {
    flex-grow: 1;
    text-align: left;
    font-weight: $font-weight-medium;
    font-size: 18px;
    margin-left: 0;
  }
}
