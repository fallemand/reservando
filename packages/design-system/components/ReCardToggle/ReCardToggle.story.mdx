import { Meta, Story, ArgsTable, Canvas, Description } from '@storybook/addon-docs/blocks';
import { ReCardToggle } from "@reservando/design-system";

<Meta title="Components/CardToggle" component={ReCardToggle} />

# ReCardToggle

<Canvas>
  <Story name="basic">
    {{
      components: { ReCardToggle },
      data() {
        return {
          checked: false,
        };
      },
      template: `
        <div style="max-width: 300px;">
          <ReCardToggle v-model="checked" label="My Label">
            My card content
          </ReCardToggle>
        </div>
      `,
    }}
  </Story>
</Canvas>

<Description of={ReCardToggle} />

## How to use

```html
<script lang="ts">
  import { ReCardToggle } from '@reservando/design-system';
</script>
```

<ArgsTable of={ReCardToggle} />
