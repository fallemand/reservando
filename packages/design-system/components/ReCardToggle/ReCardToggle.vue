<template>
  <ReCard class="re-card-toggle" :class="$attrs.class">
    <ReSwitch class="re-card-toggle__switch" :id="label.toLowerCase().replace(/\ /g, '')" :label="label" v-bind="{ ...$attrs, class: undefined }"/>
    <ReTransition transition="height">
      <div v-show="$attrs.modelValue">
        <div class="re-card-toggle__content">
          <slot />
        </div>
      </div>
    </ReTransition>
  </ReCard>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import ReCard from "../ReCard/ReCard.vue";
import ReTransition from "../ReTransition/ReTransition.vue";
import ReSwitch from "../ReSwitch/ReSwitch.vue";

const ReCardToggle = defineComponent({
  name: "ReCardToggle",
  inheritAttrs: false,
  components: {
    ReCard,
    ReTransition,
    ReSwitch,
  },
  props: {
    label: {
      type: String,
      required: true,
    },
  },
});

export default ReCardToggle;
</script>

<style lang="scss" src="./ReCardToggle.scss" />
