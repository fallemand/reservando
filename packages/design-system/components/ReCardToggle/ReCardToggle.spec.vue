<script>
import { mount } from "@vue/test-utils";
import ReCardToggle from "./ReCardToggle.vue";

describe("ReCardToggle component", () => {
  const slots = {
    default: "__CONTENT__",
  };
  const props = {
    label: "__LABEL__",
  };
  const attrs = {
    modelValue: true,
  };

  it("should render with default props", () => {
    const wrapper = mount(ReCardToggle, { props, attrs, slots });
    expect(wrapper.element).toMatchSnapshot();
  });
});
</script>
