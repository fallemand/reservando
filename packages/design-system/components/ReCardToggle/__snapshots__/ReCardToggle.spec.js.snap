// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReCardToggle component should render with default props 1`] = `
<div
  class="re-card re-card-toggle re-card-toggle"
>
  
  <div
    class="re-switch re-card-toggle__switch"
  >
    <input
      aria-checked="true"
      aria-label="checkbox"
      class="re-switch__input"
      id="__label__"
      role="switch"
      type="checkbox"
    />
    <button
      class="re-switch__control re-switch__control--checked"
      type="button"
    />
    <label
      class="re-switch__label"
      for="__label__"
    >
      __LABEL__
    </label>
  </div>
  <transition-stub>
    
    <div>
      <div
        class="re-card-toggle__content"
      >
        
        __CONTENT__
        
      </div>
    </div>
    
  </transition-stub>
  
</div>
`;
